### ISSUE TITLE
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.instabug.library.visualusersteps.c.a()' on a null object reference

### ISSUE THREAD

**Issue by zeevl** [Instabug/Instabug-React-Native#501]
By adding `Instabug.onNavigationStateChange` to `onStateChange`, I'm getting a crash at startup on Android.
```
    <NavigationContainer
       onStateChange={Instabug.onNavigationStateChange}
    >
      {renderCurrentScreen()}
      
    </NavigationContainer>
```

Results in:
```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.instabug.library.visualusersteps.c.a()' on a null object reference
    at com.instabug.library.visualusersteps.h$d.run(VisualUserStepsProvider.java:3)
    at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
    at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
    at com.instabug.library.util.threading.b$a.run(PriorityThreadFactory.java:8)
    at java.lang.Thread.run(Thread.java:764)
```

### Instabug integration code

```
  Instabug.startWithToken('...`, [
    Instabug.invocationEvent.shake,
    Instabug.invocationEvent.screenshot,
  ]);
```

### SDK Version
`instabug-reactnative@9.1.6`
`"react-native": "0.63.2",`

**Comment by AliAbdelfattah** [comment]
Hi @zeevl `Instabug.onNavigationStateChange` API is intended for react-navigation v4, we're currently working on supporting v5. Meanwhile, you can give it a try if that's possible, we'd appreciate any feedback you might have.

Snapshot:
`npm install --save https://github.com/Instabug/Instabug-React-Native#customBuild/react-navigation-v5`

New API usage:
`onStateChange={Instabug.onStateChange}`

**Comment by zeevl** [comment]
Hey, thank you for the patch.  Sorry for the delay!

One problem I've run into with this is, when hot reload happens during
development, instabug is causing an `undefined is not an object` error
accessing "state.routes":

[image: Screenshot_1599609141.png]

**Comment by zeevl** [comment]
Hey, thank you for the patch.  Sorry for the delay!

One problem I've run into with this is, when hot reload happens during
development, instabug is causing an `undefined is not an object` error
accessing "state.routes":

![Screenshot_1599609141](https://user-images.githubusercontent.com/1621969/92538608-f9e36e00-f1f3-11ea-84a5-7652ffab13fc.png)

**Comment by AliAbdelfattah** [comment]
Hmm.. let me look into that.

**Comment by zeevl** [comment]
It only happens on occasion.. seems to be a timing issue.   Otherwise the
repo steps work great!

**Comment by AliAbdelfattah** [comment]
@zeevl Fixed in the official release v9.1.8. Please give it a try and let us know if you encounter any other issues.
